data/test/scenarios/wml_tests/ScenarioWML/EventWML/ActionWML/DirectActionsWML/kill_fires_events.cfg

```diff
@@ -0,0 +1,34 @@
+# This test checks if the last_breath and die events are triggered when a unit on the recall list is killed with [kill] tag
+# The default behavior is that only units on the map trigger the events, so this test will fail with alice_breath and alice_die
+{GENERIC_UNIT_TEST "kill_fires_events" (
+    [event]
+        name=last_breath
+		[set_variable]
+			name=$unit.id|_breath
+			value=yes
+		[/set_variable]
+    [/event]
+    [event]
+        name=die
+		[set_variable]
+			name=$unit.id|_die
+			value=yes
+		[/set_variable]
+    [/event]
+    [event]
+        name=start
+		[put_to_recall_list]
+			id=alice
+		[/put_to_recall_list]
+		[kill]
+			id=alice,bob
+			fire_event=yes
+		[/kill]
+        {ASSERT ({VARIABLE_CONDITIONAL bob_die equals yes})}
+        {ASSERT ({VARIABLE_CONDITIONAL alice_die equals yes})}
+        {ASSERT ({VARIABLE_CONDITIONAL bob_breath equals yes})}
+        {ASSERT ({VARIABLE_CONDITIONAL alice_breath equals yes})}

```

Comment:
When possible, it's best to try and have the test reach the SUCCEED macro, so if alice isn't actually going to trigger the events, changing those from checking `equals` to `not_equals` would be better.

Reply:
For `yes` and `no`, there's `boolean_equals` and `boolean_not_equals`. The difference from`equals` is that `boolean_equals no` will also match the variable being unitialised or `false` as well as `no` - and in this case, matching when it's uninitialised matches if the event didn't fire.

Reply:
That is quite useful. It would allow me to abbreviate some code. I hadn't realized that, thanks, I thought they were equivalent.